images/linux/scripts/installers/azure-devops-cli.sh

```diff
@@ -4,12 +4,31 @@
 ##  Desc:  Installed Azure DevOps CLI (az devops)
 ################################################################################
 
-# AZURE_EXTENSION_DIR shell variable defines where modules are installed
-# https://docs.microsoft.com/en-us/cli/azure/azure-cli-extensions-overview
-export AZURE_EXTENSION_DIR=/opt/az/azcliextensions
-echo "AZURE_EXTENSION_DIR=$AZURE_EXTENSION_DIR" | tee -a /etc/environment
+source $HELPER_SCRIPTS/etc-environment.sh
+export AZURE_DEVOPS_EXT_CONFIG_DIR=/opt/az-devops/config
+setEtcEnvironmentVariable "AZURE_DEVOPS_EXT_CONFIG_DIR" "$AZURE_DEVOPS_EXT_CONFIG_DIR"
+
+export AZURE_DEVOPS_CACHE_DIR=/opt/az-devops/config/cache
+setEtcEnvironmentVariable "AZURE_DEVOPS_CACHE_DIR" "$AZURE_DEVOPS_CACHE_DIR"
+
+sudo mkdir /opt/az-devops
 
 # install azure devops Cli extension
 az extension add -n azure-devops
 
-invoke_tests "CLI.Tools" "Azure DevOps CLI"
\ No newline at end of file
+# calling az devops login to force it to install `keyring`. Login will actually fail, redirecting error to null
+echo "dummy" | az devops login > /dev/null
+# calling az devops logout to be sure no credentials remain.
+az devops logout > /dev/null
+
+# warm-up az devops
+az devops --help > /dev/null
+az pipelines --help > /dev/null
+az boards --help > /dev/null
+az repos --help > /dev/null
+az artifacts --help > /dev/null
+
+invoke_tests "CLI.Tools" "Azure DevOps CLI"
+
+chown -R "$SUDO_USER:$SUDO_USER" "/opt/az"
+chown -R "$SUDO_USER:$SUDO_USER" "/opt/az-devops"
```

Comment:
The same as above: I suppose it may be better to keep tests invocation at the very end of the script.

Reply:
The tests may cause files to be generated in these directories, moving the test execution above would cause these files to have the wrong mods.

Ideally the chmod would move to the post-reboot script

Reply:
It looks like `cleanup.sh` or `reboot.sh` would currently be the ideal locations.

Reply:
Hey, we discussed it internally in our team and decided to accept this for a while. Therefore let's close this discussion.